### PR TITLE
Add schema hash id

### PR DESCRIPTION
**Pull request by cafca**
- adds a restriction that schema names cannot end with an underscore

### CODE REVIEW

**Comment by cafca** [comment]
Sorry for the review request, this should be merged after #162.

File: website/docs/01-writing-data/schemas.md [hunk 1]
```diff
@@ -87,6 +87,14 @@ id: schemas
   - the first section is the name, which must have 1-64 characters, must start with a letter and must contain only alphanumeric characters and underscores
   - the remaining sections are the document view id of the schema's `schema_definition_v1` document, represented as alphabetically sorted hex-encoded operation ids, separated by underscores.
   - example `profile_picture_0020c65567ae37efea293e34a9c7d13f8f2bf23dbdc3b5c7b9ab46293111c48fc78b`
+- as application schema ids can potentially grow very large the _schema hash id_ is an alterantive identifier with limited size
+  - it's possible for an adversary to generate schemas with a very large schema id, although this is not expected in regular use
+  - applications should choose to identify schemas by their _schema hash id_ when this fact can be exploited or a short identifier results in better usability
```
Comment (cafca) [inline website/docs/01-writing-data/schemas.md:92]: I think we should consider limiting the length of document view ids to some large number to not make this so exploitable.

**Comment by cafca** [comment]
Problem: as a schema name can end with an underscore, using two underscores to differentiate schema hash ids from schema ids doesn't really work :D

**Comment by sandreae** [comment]
Maybe option B is easiest and shouldn't cause many problems for people naming schemas?

**Comment by cafca** [comment]
I like that, do you agree? @adzialocha

**Comment by adzialocha** [comment]
> I like that, do you agree? @adzialocha

Yes, I like that too! :+1:

**Review by sandreae — APPROVED** [review]
Looks good to me.

Only one niggle, which is not worth blocking this merge for as I'm not sure it's a problem... but, using the double underscore as the separator for hash ids maybe a little hard to spot if we're debugging (for example). I could imagine not spotting that and being unsure if I'm looking at a regular schema id or a hash id. But maybe let's see where we use these ids first and then think again.

**Comment by cafca** [comment]
Yes, that's a good point to evaluate - maybe before publishing 1.0.0? https://github.com/p2panda/handbook/issues/171

**Pull request merged**